engine/robot.py

```diff
@@ -1,45 +1,14 @@
 import threading
```

Comment:
just an overall comment, but we shouldn't have files that are like 500+ lines, im wondering if there's a way we can split this up more? like maybe have a file for like robot_commands that handles moving forward etc. 

idk but this file seems tooo clunky

Reply:
agreed! ill add it to the list of new tasks -- think we could further refactor robot aside from splitting out these attributes to robot_state